integration/run/run_test.go

```diff
@@ -1574,6 +1575,9 @@ func TestEnforcedQuota(t *testing.T) {
 	}
 
 	// Update the project to enforce quota.
+	if err := kclient.Get(ctx, types.NamespacedName{Name: project.Name}, project); err != nil {
```

Comment:
I think the update will still fail if a controller updates the project between this get and the following update call. With that in mind, I think we'll want to either:
1. switch to an apply
2. perform a get+update repeatedly until you're successful

The later I had to do for some e2e tests for events:
```go
	app = helper.WaitForObject(t, helper.Watcher(t, c), &apiv1.AppList{}, app, func(obj *apiv1.App) bool {
		obj.Spec.Stop = &stop
		return kc.Update(ctx, obj) == nil
	})
```

Reply:
Thanks! I went with the `Wait` approach. Its up now 👍🏻